src/main/java/core/basesyntax/service/impl/CsvReporter.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Reporter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CsvReporter implements Reporter {
+    private static final String HEADER = "fruit,quantity";
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public List<String> generate(Map<String, Integer> finalState) {
+        List<String> report = new ArrayList<>();
+        report.add(HEADER);
+        finalState.entrySet().stream()
+                .map(e -> e.getKey() + SEPARATOR + e.getValue())
+                .forEach(report::add);
```

Comment:
I think you can use collector and collect to list

Reply:
I actually need to add header before so i don't really know how to do it by stream